### DISCUSSION TITLE
Tenants

### DISCUSSION THREAD

**Discussion by ChuckMoe** [fief-dev/fief#47] [Q&A] [answered]
I have some questions regarding the tenant system.

1. The documentation states that you are currently unable to create tenants via the dashboard. How can you create a new tenant than? I haven't found any kind of API documentation regarding this.
2. I have the following scenario: Big customers should be able to have multiple accounts with at least one account that can manage the others. Is this possible with the tenant system or would you recommend to implement this via user fields?

Cheers
Moe

**Comment by frankie567** [comment, marked as answer]
Hi @ChuckMoe :wave:

1. Yes, currently there is no built-in way to create a new tenant. But I'll fix this soon 😃 
2. Just to be sure we are talking about the same things:
    * When you talk about "account", you talk about a user account, with an e-mail and a password, right?
    * When you say "manage the others", what type of actions do you have in mind?

**Reply by ChuckMoe** [reply]
Hi @frankie567 :)

1.  I'm looking forward to this. I really like your project(s)!
2. With "account", I mean a user account, that is true. In essence, this 'manager account' needs the ability to 'invite' users into his group. Let me give you an example:

I provide a service. Users can register normally. Now an organization with multiple offices becomes a customer of me. They would need a user account for every office location that wants to use our service then. I would like to group all of their accounts. In this group of account's there has to be one that is able to invite/add new accounts to their group.

I thought that this is maybe possible using the tenant system. I could give the organization their tenant admin access. Then they can add and manage all of their own user accounts. In this case, an option to disable the registration process for specific tenants would be great, so that no one can accidentally join this tenant.

**Reply by frankie567** [reply]
Thank you for those details! I know see what you want to do.

However, I don't think tenants would be a good candidate for what you want to do. Tenants are designed to isolate users: they can't be moved from one tenant to another. Authentication Clients are also tenant-specific. Put another way, Tenants act as completely isolated apps.

What you describe looks more like a group system, where users can be added to groups, and some users can manage/invite users of this group. Potentially, users can belong to several groups. This is something I would also like to add in Fief, ultimately.

For now, I think you'll need to implement this logic yourself. You can define a *Group* entity in your app and store the list of User IDs that belong to this group.

**Reply by ChuckMoe** [reply]
Thanks for the clarification! I wasn't aware of how much the tenants were isolated.

Then I do it this way :+1:

**Reply by frankie567** [reply]
Hello @ChuckMoe 👋 

Just for you to know: the latest version, [`v0.16.7`](https://github.com/fief-dev/fief/releases/tag/v0.16.7), includes the possibility to create new tenant from the admin API or dashboard!

Cheers!
